cmake/StaticAnalyzers.cmake

```diff
@@ -85,3 +85,36 @@ macro(enable_clang_Format)
         message(FATAL_ERROR "clang-format requested but executable not found")
     endif ()
 endmacro()
+
+# Create a custom target that executes cmake-lint.
+#
+# This is set as a build dependency of other targets.
+#
+# Linter configuration is in cmake-format.yaml.
+macro(enable_cmake_lint)
+    find_program(OPENASSETIO_CMAKELINT_EXE NAMES cmake-lint)
+    if (OPENASSETIO_CMAKELINT_EXE)
+        file(
+            GLOB_RECURSE _sources
+            LIST_DIRECTORIES false
+            CONFIGURE_DEPENDS # Ensure we re-scan if files change.
+            ${PROJECT_SOURCE_DIR}/cmake/*.cmake
+            ${PROJECT_SOURCE_DIR}/src/CMakeLists.txt
```

Comment:
[question] (Sorry CMake noob) Do we need to trigger the rescan if the other nested `CMakeLists.txt` files too?

Reply:
Not sure what you mean by trigger the rescan. GLOB_RECURSE is weird in that the last component of a file path (i.e. the file name) is searched for recursively under the given prefix (i.e. everything up to the file name).  So the nested `CMakeLists.txt` are discovered.  I'll break one of those deliberately to demonstrate.

Reply:
Added a violation, see https://github.com/OpenAssetIO/OpenAssetIO/actions/runs/6889451328/job/18740464053?pr=1208

Reply:
Oh that is confusing, even without any wild cards? Their docs dont make that particularly clear.... but makes sense, thanks.